firebase-sessions/src/main/kotlin/com/google/firebase/sessions/SharedSessionRepository.kt

```diff
@@ -153,17 +155,26 @@ constructor(
             "Notified ${subscriber.sessionSubscriberName} of new session $sessionId"
           NotificationType.FALLBACK ->
             "Notified ${subscriber.sessionSubscriberName} of new fallback session $sessionId"
-        }
+        },
       )
     }
   }
 
   private fun shouldInitiateNewSession(sessionData: SessionData): Boolean {
-    sessionData.backgroundTime?.let {
-      val interval = timeProvider.currentTime() - it
-      return interval > sessionsSettings.sessionRestartTimeout
+    sessionData.backgroundTime?.let { backgroundTime ->
+      val interval = timeProvider.currentTime() - backgroundTime
+      if (interval > sessionsSettings.sessionRestartTimeout) {
+        // Passed session restart timeout, so should initiate a new session
+        return true
+      }
     }
-    Log.d(TAG, "No process has backgrounded yet, should not change the session.")
```

Comment:
nit: Logs for cold start and background time.

Reply:
Done